SimPEG/regularization/vector.py

```diff
@@ -924,33 +922,56 @@ def f_m(self, m):
     def f_m_deriv(self, m) -> csr_matrix:
         r"""Derivative of the regularization kernel function.
 
-        For first-order smoothness regularization in the x-direction, the derivative of the
-        regularization kernel function with respect to the model is given by:
+        See :func:`base.SmoothnessFirstOrder.f_m_deriv` for more information.
+        """
+        return self.cell_gradient
```

Comment:
if `f(m) = cell_gradient @ amplitude(m)`, why do we do not see the amplitude derivative in `f_m_deriv` and only cell_gradient as if `f_m` was linear? What do I miss?

Reply:
because we measure amplitude**2, ends up being the sum of gradients of components square. So we really don't gain much in L2, only does something different in the Lp